common/src/main/java/net/adoptopenjdk/icedteaweb/StringUtils.java

```diff
@@ -67,4 +69,36 @@ public static boolean isBlank(final String str) {
         lines.add(tmp);
         return lines;
     }
+
+    /**
+     * Checks whether the first part of the given prefixString is a prefix for any of the strings
+     * in the specified array.
+     * If the {@code prefixString} contains multiple words separated by a space character, the
+     * first word is taken as prefix for comparison.
+     *
+     * @param prefixString the prefixString string
+     * @param available the strings to test
+     * @return true if the first part of the given prefixString is a prefix for any of the strings
+     * in the specified array or the specified array is empty or null, false otherwise
+     */
+    public static boolean hasPrefixMatch(final String prefixString, final String[] available) {
+        Assert.requireNonBlank(prefixString, "prefixString");
+
+        if (available == null || available.length == 0){
+            return true;
```

Comment:
from my understanding of the javadoc this should return `false`

Reply:
you are right, Javadoc does not explicitly say this (only in the @return text). I've improved doc for this. It was also my first reaction when I saw the code that I wanted the method ti return false here. But after studying the existing test cases it looks like this behavior was by intention.  And I didn't want to change functionality here with this refactoring. So for now I would keep it like this until we have more confidence where to check for side effects when making this method more "logic".